core/string/ustring.cpp

```diff
@@ -1644,6 +1644,17 @@ String String::hex_encode_buffer(const uint8_t *p_buffer, int p_len) {
 	return ret;
 }
 
+Vector<uint8_t> String::hex_decode() const {
+	const String *s = this;
+	Vector<uint8_t> out;
+	int len = s->size() / 2;
```

Comment:
Probably, in case of an odd length string, the 0th character should be treated as 4 bits of the 0th byte?

Reply:
Can you give me a example?

Reply:
I think `"5f1f2".hex_decode()` should return `PackedByteArray([0x05, 0xF1, 0xF2])`. Or maybe `PackedByteArray([0x5F, 0x1F, 0x20])`, I'm not sure. Or at least it should print the error "An even length string was expected".

Reply:
I believe i had a even length check a while ago, but i remvoed it because hex_encode produced non even strings. ill double check.